src/lib/modules/scaffolding/framework/react/index.ts

```diff
@@ -0,0 +1,101 @@
+
+export class React implements Builder {
+  constructor(private embark: Embark,
+              private description: Description,
+              private contracts: Contract[],
+              private options: Options) {
+  }
+
+  public readonly build = async () => {
+    await this.installDependencies();
+
+    return [].concat.apply([], Object.keys(this.description.data).map((contractName) => {
+      const [indexCode, dappCode] = this.generateCodes(contractName);
+      if (indexCode && dappCode) {
+        const files = this.saveFiles(contractName, indexCode, dappCode);
+        this.updateEmbarkJson(contractName, files);
+        return files;
+      } else {
+        return [];
+      }
+    }));
+  }
+
+  private readonly updateEmbarkJson = (contractName: string, files: string[]) => {
+    const embarkJsonPath = path.join(fs.dappPath(), "embark.json");
+    const embarkJson = fs.readJSONSync(embarkJsonPath);
+    embarkJson.app[`js/${contractName}.js`] = `app/${contractName}.js`;
+    embarkJson.app[`${contractName}.html`] = `app/${contractName}.html`;
+
+    fs.writeFileSync(embarkJsonPath, JSON.stringify(embarkJson, null, 2));
+  }
+
+  private readonly generateCodes = (contractName: string) => {
+    const indexSource = fs.readFileSync(indexTemplatePath, "utf-8");
+    const dappSource = fs.readFileSync(dappTemplatePath, "utf-8");
+
+    const indexTemplate = Handlebars.compile(indexSource);
+    const dappTemplate = Handlebars.compile(dappSource);
+
+    const indexData = {
+      filename: contractName.toLowerCase(),
+      title: contractName,
+    };
+
+    const contract = this.contracts.find((c) => c.className === contractName);
+    if (!contract) {
+      return [];
+    }
+
+    const dappData = {
+      contractName,
+      functions: contract.abiDefinition.filter((entry) => entry.type === "function"),
+    };
+
+    return [indexTemplate(indexData), dappTemplate(dappData)];
+  }
+
+  private readonly installDependencies = () => {
+    const cmd = "npm install react react-bootstrap react-dom";
+    return new Promise<void>((resolve, reject) => {
+      utils.runCmd(cmd, null, (error: string) => {
+        if (error) {
+          return reject(new Error(error));
+        }
+
+        resolve();
+      });
+    });
+  }
+
+  private readonly saveFiles = (contractName: string, indexCode: string, dappCode: string) => {
+    const indexFilePath = path.join(fs.dappPath(), "app", `${contractName}.html`);
+    const dappFilePath = path.join(fs.dappPath(), "app", `${contractName}.js`);
+
+    if (!this.options.overwrite && (fs.existsSync(indexFilePath) || fs.existsSync(dappFilePath))) {
+      return [];
+    }
+
+    fs.writeFileSync(indexFilePath, indexCode);
+    fs.writeFileSync(dappFilePath, dappCode);
+
+    this.embark.logger.info(__(`${indexFilePath} generated`));
+    this.embark.logger.info(__(`${dappFilePath} generated`));
+    return [indexFilePath, dappFilePath];
+  }
+}
```

Comment:
Very solid again! but as mentioned above, I'd expect those property functions to be methods on the class.